src/Microsoft.TemplateEngine.Edge/Installers/NuGet/NuGetVersionHelper.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using NuGet.Versioning;
+
+namespace Microsoft.TemplateEngine.Edge.Installers.NuGet
+{
+    internal static class NuGetVersionHelper
+    {
+        public static bool IsSupportedVersionString(string? versionString)
+        {
+            return
+                string.IsNullOrEmpty(versionString)
+                ||
+                NuGetVersion.TryParse(versionString, out _)
+                ||
+                FloatRange.TryParse(versionString, out _);
+        }
+
+        /// <summary>
+        /// Tries to parse given string and return <see cref="FloatRange"/> provided there
+        ///  is an existing floating range behavior in the provided string.
+        ///  null or empty string are regarded to be requests to any release version (behavior identical to '*').
+        /// </summary>
+        /// <param name="versionString">Input string to be parsed.</param>
+        /// <param name="floatRange">Output <see cref="FloatRange"/> parameter, populated in case function returned true.</param>
+        /// <returns></returns>
+        public static bool TryParseFloatRangeEx(string? versionString, out FloatRange floatRange)
+        {
+            floatRange =
+                string.IsNullOrEmpty(versionString) ?
+                new FloatRange(NuGetVersionFloatBehavior.Major) : FloatRange.Parse(versionString);
```

Comment:
can `FloatRange.Parse` throw in case of invalid version?
In this case, try-pattern is not followed.

Reply:
`FloatRange.Parse` just calls `FloatRange.TryParse`:
https://github.com/NuGet/NuGet.Client/blob/bc4b41f3ca294ccc8aedc8af5159c215943e1ed2/src/NuGet.Core/NuGet.Versioning/FloatRange.cs#L167

so it's not throwing and change in behavior would be a breaking change that would broke others as well.